src/input.cpp

```diff
@@ -339,35 +338,25 @@ void update_fish_color_support(void) {
 
 int input_init() {
     if (is_init) return 1;
-
     is_init = true;
-
     input_common_init(&interrupt_handler);
 
-    const env_var_t term = env_get_string(L"TERM");
     int errret;
-    if (setupterm(const_cast<char *>(wcs2string(term).c_str()), STDOUT_FILENO, &errret) == ERR) {
-        debug(0, _(L"Could not set up terminal"));
-        if (errret == 0) {
-            debug(0, _(L"Check that your terminal type, '%ls', is supported on this system"),
-                  term.c_str());
-            debug(0, _(L"Attempting to use '%ls' instead"), DEFAULT_TERM);
-            env_set(L"TERM", DEFAULT_TERM, ENV_GLOBAL | ENV_EXPORT);
-            const std::string default_term = wcs2string(DEFAULT_TERM);
-            if (setupterm(const_cast<char *>(default_term.c_str()), STDOUT_FILENO, &errret) ==
-                ERR) {
-                debug(0, _(L"Could not set up terminal"));
-                exit_without_destructors(1);
-            }
-        } else {
+    if (setupterm(NULL, STDOUT_FILENO, &errret) == ERR) {
+        env_var_t term = env_get_string(L"TERM");
+        debug(1, _(L"Preferred terminal type '%ls' is not valid"), term.c_str());
+        debug(1, _(L"Check that your terminal type is supported on this system"));
+        env_set(L"TERM", DEFAULT_TERM, ENV_GLOBAL | ENV_EXPORT);
+        if (setupterm(NULL, STDOUT_FILENO, &errret) == ERR) {
+            debug(0, _(L"Unable to setup terminal using your preferred type or the '%ls' fallback"),
+                  DEFAULT_TERM);
             exit_without_destructors(1);
+        } else {
+            debug(1, _(L"Using fallback terminal type '%ls' instead"), DEFAULT_TERM);
         }
```

Comment:
I was viewing this as a warning since we fell back to something useful that should work reasonably well for most people but I think you're right -- this should be at error level since the user made an error.